src/blueprints/Do.Blueprints.Service.Application/Business/Default/DefaultBusinessFeature.cs

```diff
@@ -11,60 +11,133 @@ namespace Do.Business.Default;
 public class DefaultBusinessFeature(List<Assembly> _domainAssemblies)
     : IFeature<BusinessConfigurator>
 {
-    const BindingFlags _defaultMemberBindingFlags = BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.DeclaredOnly;
+    const BindingFlags _bindingFlags = BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.DeclaredOnly;
 
     public void Configure(LayerConfigurator configurator)
     {
-        configurator.ConfigureDomainAssemblyCollection(assemblies =>
+        configurator.ConfigureDomainTypeCollection(types =>
         {
             foreach (var assembly in _domainAssemblies)
             {
-                assemblies.Add(assembly);
+                types.AddFromAssembly(assembly, except: type =>
+                    (type.IsSealed && type.IsAbstract) || // if type is static
+                    type.IsAssignableTo(typeof(Exception)) ||
+                    type.IsAssignableTo(typeof(Attribute)) ||
+                    type.IsAssignableTo(typeof(Delegate))
+                );
             }
         });
 
         configurator.ConfigureDomainBuilderOptions(options =>
         {
-            options.ConstuctorBindingFlags = _defaultMemberBindingFlags;
-            options.MethodBindingFlags = _defaultMemberBindingFlags;
-            options.PropertyBindingFlags = _defaultMemberBindingFlags;
+            options.BindingFlags.Constructor = _bindingFlags;
+            options.BindingFlags.Method = _bindingFlags;
+            options.BindingFlags.Property = _bindingFlags;
+
+            options.BuildLevels.Add(context => context.DomainTypesContain(context.Type), BuildLevels.Members);
+            options.BuildLevels.Add(context => context.Type.IsGenericType && context.DomainTypesContain(context.Type.GetGenericTypeDefinition()), BuildLevels.Members);
+            options.BuildLevels.Add(type => !type.IsValueType, BuildLevels.Inheritance);
+            options.BuildLevels.Add(type => type.IsGenericType, BuildLevels.Generics);
+        });
+
+        configurator.ConfigureDomainIndexOptions(options =>
+        {
+            options.Type.Add(AttributeIndex.New<ServiceAttribute>());
+            options.Type.Add(AttributeIndex.New<TransientAttribute>());
+            options.Type.Add(AttributeIndex.New<ScopedAttribute>());
+            options.Type.Add(AttributeIndex.New(new SingletonAttribute()));
+            options.MethodGroup.Add(AttributeIndex.New<ApiMethodAttribute>());
```

Comment:
it is better when you help while creating an object through builder extensions. E.g., `options.Type.Add<ServiceAttribute>()` is better than `options.Type.Add(new AttributeIndex<ServiceAttribute>())` which is better than `options.Type.Add(AttributeIndex.New<ServiceAttribute>())`.

it is hard for us to remember how an instance of a custom type was created. first you look for the simplest overload of the method, if not you try to create an instance through available constructors. but when you don't even have a constructor option, you don't look for static factory methods right away, instead of that you mostly give up trying and lean towards documentation, which is bad for a developer experience

Reply:
We decided to remove `AttributeIndex` and only use attribute type for indices